packages/features/bookings/Booker/components/AvailableTimeSlots.tsx

```diff
@@ -21,20 +23,84 @@ type AvailableTimeSlotsProps = {
  * will also fetch the next `extraDays` days and show multiple days
  * in columns next to each other.
  */
-export const AvailableTimeSlots = ({ extraDays, limitHeight, seatsPerTimeSlot }: AvailableTimeSlotsProps) => {
-  const selectedDate = useBookerStore((state) => state.selectedDate);
-  const setSelectedTimeslot = useBookerStore((state) => state.setSelectedTimeslot);
-  const setSeatedEventData = useBookerStore((state) => state.setSeatedEventData);
+export const AvailableTimeSlots = ({
+  extraDays,
+  limitHeight,
+  seatsPerTimeslot,
+  eventSlug,
+}: AvailableTimeSlotsProps) => {
   const event = useEvent();
-  const date = selectedDate || dayjs().format("YYYY-MM-DD");
   const containerRef = useRef<HTMLDivElement | null>(null);
+  const [
+    selectedDate,
+    selectedDatesAndTimes,
+    setSelectedDatesAndTimes,
+    setSelectedTimeslot,
+    setSeatedEventData,
+  ] = useBookerStore(
+    (state) => [
+      state.selectedDate,
+      state.selectedDatesAndTimes,
+      state.setSelectedDatesAndTimes,
+      state.setSelectedTimeslot,
+      state.setSeatedEventData,
+    ],
+    shallow
+  );
+
+  const date = selectedDate || dayjs().format("YYYY-MM-DD");
 
   const onTimeSelect = (
     time: string,
     attendees: number,
     seatsPerTimeSlot?: number | null,
     bookingUid?: string
   ) => {
+    // Used for selecting multiple time slots and assigning the selected values to a date
+    if (eventSlug) {
+      if (selectedDatesAndTimes && selectedDatesAndTimes[eventSlug]) {
+        const selectedDatesAndTimesForEvent = selectedDatesAndTimes[eventSlug];
+        const selectedSlots = selectedDatesAndTimesForEvent[selectedDate as string] ?? [];
+        if (selectedSlots?.includes(time)) {
+          // Checks whether a user has removed all their timeSlots and thus removes it from the selectedDatesAndTimesForEvent state
+          if (selectedSlots?.length > 1) {
+            const updatedDatesAndTimes = {
+              ...selectedDatesAndTimes,
+              [eventSlug]: {
+                ...selectedDatesAndTimesForEvent,
+                [selectedDate as string]: selectedSlots?.filter((slot: string) => slot !== time),
+              },
+            };
+
+            setSelectedDatesAndTimes(updatedDatesAndTimes);
+          } else {
+            const updatedDatesAndTimesForEvent = { ...selectedDatesAndTimesForEvent };
+            delete updatedDatesAndTimesForEvent[selectedDate as string];
+            setSelectedTimeslot(null);
+            setSelectedDatesAndTimes({ ...selectedDatesAndTimes, [eventSlug]: updatedDatesAndTimesForEvent });
+          }
+          return;
+        }
```

Comment:
Here we're checking whether the user is clicking on an already selected time slot thus means that they want to un-select it, so we're checking whether the length of available time slots after removing this will be 0 or not, and if it will become 0 we would remove that selected dates from the selectedDatesAndTimes for this event.